### PR TITLE
Fix issue with debug command not running in docker properly

### PR DESCRIPTION
**Pull request by fkuatspyro**
- Fix not running debug in docker if docker environment
- Add unit tests for debug
- Bump version

### CODE REVIEW

File: tests/ut/data/scargo_test_esp32.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ homepage-url = "www.hello-world.com"
 
 bin_name = "esp32proj"
 target = "esp32"
-build-env = "docker"
+build-env = "native"
```
Comment (aaksenczuk) [inline tests/ut/data/scargo_test_esp32.toml:9]: please keep docker as default

File: tests/ut/data/scargo_test_x86.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ homepage-url = "www.hello-world.com"
 
 bin_name = "testproject"
 target = "x86"
-build-env = "docker"
+build-env = "native"
```
Comment (aaksenczuk) [inline tests/ut/data/scargo_test_x86.toml:9]: please keep docker as default

File: tests/ut/ut_scargo_debug.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    bin_path.parent.mkdir(parents=True)
+    bin_path.touch()
+
+    fp.register(["/usr/bin/which", "openocd"], stdout="openocd")
+    fp.register(["sudo", "openocd", "-f", ".devcontainer/openocd-script.cfg"])
+    fp.register(
+        [
+            "gdb-multiarch",
+            bin_path.absolute(),
+            "--eval-command=target extended-remote localhost:3333",
```
Comment (aaksenczuk) [inline tests/ut/ut_scargo_debug.py:58]: target extended-remote localhost:3333 
this is something which repeat over all uC. I will suggest to extract it as a const e.g. GDB_SETTINGS?

File: tests/ut/data/scargo_test_esp32.toml (hunk 1 above)
Comment (fkuatspyro) [reply tests/ut/data/scargo_test_esp32.toml:9]: The reason why I changed it to _native_ is that these tomls are used in tests only and I don't want to trigger scargo to run again same command in docker.

**Review by aaksenczuk — APPROVED** [review]

**Pull request merged**